### PR TITLE
Don't ForceNew on certificate_arn for resourceAwsCognitoUserPoolDomain

### LINKED ISSUE #998: aws.cognito.UserPoolDomain changing certificateArn tries to replace
## Version
Windows - v2.3.0

## Description

If the certificate arn of a UserPoolDomain changes, the provider tries to replace the domain by deleting it and then recreating it.
This runs into all kind of issues, as even after deletion, the underlying (hidden) CloudFront that AWS use will still exist, preventing recreation of the UserPoolDomain until it has been destroyed.
This is also an issue when the provider tries to replace a certificate, and decides to delete the old one that is in use.

In the AWS console, the user pool domain's certificate can easily be changed without deleting and recreating the domain.

## Error

```
Diagnostics:
  aws:cognito:UserPoolDomain (domain):
    error: Error creating Cognito User Pool Domain: InvalidParameterException: Domain already exists.
```

### PR DESCRIPTION
**Pull request by leezen**
Fixes https://github.com/pulumi/pulumi-aws/issues/998

### CODE REVIEW

**Comment by lukehoban** [comment]
This resource doesn’t appear to define any Update implementation at all currently - which only works if all inputs are ForceNew.  Pretty sure an Update function will be needed to handle update of this certificate_arn property.

**Comment by leezen** [comment]
Ah, darnit -- thanks for pointing that out. Let me move this PR to WIP, then and look at actually implementing that.

**Comment by leezen** [comment]
I should probably note that I haven't really touched the providers before, so please give this a vigorous look. I think I understood what needed to be done here, but I could be totally wrong.

File: aws/resource_aws_cognito_user_pool_domain.go [hunk 1]
```diff
@@ -163,32 +140,91 @@ func resourceAwsCognitoUserPoolDomainDelete(d *schema.ResourceData, meta interfa
 		return err
 	}
 
+	err = waitForUserPoolDomainDelete(conn, d.Id())
+	return err
+}
+
+func resourceAwsCognitoUserPoolDomainUpdate(d *schema.ResourceData, meta interface{}) error {
```
Comment (stack72) [inline aws/resource_aws_cognito_user_pool_domain.go:147]: There needs to be a check in here to see if the `certificate_arn` has changed

it's usually along the lines of

```
if d.HasChange("certificate_arn")
```

**Comment by stack72** [comment]
You can run the tests for this resource using the following:

```
TF_ACC=1 go test -v ./aws -run=TestAccAWSCognitoUserPoolDomain_
```

That will run all of the CognitoUserPoolDomain tests

**Comment by stack72** [comment]
@leezen 1 more thing, we don't use pulumi-master branch - we have a branch per release

The latest branch would be `upstream-v2.65.0` so if you send a PR into that branch, then it would be better :)

File: aws/resource_aws_cognito_user_pool_domain.go (hunk 1 above)
Comment (leezen) [reply aws/resource_aws_cognito_user_pool_domain.go:147]: Interesting -- I see this pattern used in some, but not all of the resources, e.g. the domain pool doesn't do this. Seems better to check, so I'll definitely do that.

**Comment by leezen** [comment]
Closing to modify the PR to go against the versioned branch

**Pull request closed without merging**